isabl_cli/batch_systems/slurm.py

```diff
@@ -122,41 +122,56 @@ def submit_slurm_array(
                 # submit a dependency job on failure
                 # important when the scheduler kills the head job
                 dependency = "${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}"
-                afternotok = (
+                after_not_ok_job = (
                     f"sbatch {extra_args} --depend=afternotok:{dependency} --kill-on-invalid-dep yes "
-                    f'-o {join(rundir, "head_job.exit")} -J "EXIT: {dependency}" '
+                    f'--export=TMP,TMPDIR,TMP_DIR -o {join(rundir, "head_job.exit")} -J "EXIT: {dependency}" '
                     f"<< EOF\n#!/bin/bash\n{exit_command}\nEOF\n"
                 )
 
                 # use random sleep to avoid parallel API hits
                 f.write(
-                    f"#!/bin/sh\nsleep {random.uniform(0, 10):.3} && "
-                    f"({afternotok}) && bash {command}"
+                    f"#!/bin/bash\n\n"
+                    f"sleep {random.uniform(0, 10):.3} && "
+                    f"({after_not_ok_job}) && bash {command}"
                 )
 
-            for j in "log", "err", "exit":
+            for j in "log", "err", "exit", "slurm":
                 src = join(rundir, f"head_job.{j}")
                 dst = join(root, f"{j}.{index}")
                 open(src, "w").close()
                 utils.force_symlink(src, dst)
 
     with open(join(root, "in.sh"), "w") as f:
-        f.write(f"#!/bin/sh\nbash {root}/in.$SLURM_ARRAY_TASK_ID")
```

Comment:
Is the second bash here redundant?

Reply:
One is for the shebang. The other one is for execute the file.

```bash
#!/bin/bash

bash <command>
```

I guess it could work without it. as long as the file `{root}/in.$SLURM_ARRAY_TASK_ID` has the proper shebang too.